pytensor/link/jax/dispatch/tensor_basic.py

```diff
@@ -205,3 +211,39 @@ def tri(*args):
         return jnp.tri(*args, dtype=op.dtype)
 
     return tri
+
+
+@node_rewriter([AllocDiag])
+def eagerly_inline_alloc_diag(fgraph, node):
+    """
+    Inline `AllocDiag` OpFromGraph into the graph so the component Ops can themselves be jaxified
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        The function graph being rewritten
+    node: Apply
+        Node of the function graph to be optimized
+
+    Returns
+    -------
+
+    """
+    [input] = node.inputs
+    [output] = node.op.inner_outputs
+    inner_input = output.owner.inputs[1]
+
+    inline = pytensor.clone_replace(output, {inner_input: input})
+
+    return [inline]
+
+
+remove_alloc_ofg_opt = SequenceDB()
+remove_alloc_ofg_opt.register(
+    "inline_alloc_diag",
+    in2out(eagerly_inline_alloc_diag),
+    "jax",
+)
+
+# Do this right away so other JAX rewrites can act on the inner graph
+optdb.register("jax_inline_alloc_diag", remove_alloc_ofg_opt, "jax", position=0)
```

Comment:
May sound silly but I would put this in a `tensor/rewriting/jax.py` so we know it's specific to that backend